plugin/evm/vm.go

```diff
@@ -1221,15 +1229,41 @@ func (vm *VM) setAppRequestHandlers() {
 			},
 		},
 	)
+	if err := vm.RegisterLeafRequestHandler(message.StateTrieNode, "sync_state_trie_leaves", evmTrieDB, message.StateTrieKeyLength, true); err != nil {
+		return fmt.Errorf("failed to register leaf request handler for state trie: %w", err)
+	}
+	// Register atomic trieDB for serving atomic leafs requests.
+	if err := vm.RegisterLeafRequestHandler(atomicsync.AtomicTrieNode, "sync_atomic_trie_leaves", vm.atomicTrie.TrieDB(), atomicstate.AtomicTrieKeyLength, false); err != nil {
+		return fmt.Errorf("failed to register leaf request handler for atomic trie: %w", err)
+	}
+
 	networkHandler := newNetworkHandler(
 		vm.blockChain,
 		vm.chaindb,
-		evmTrieDB,
-		vm.atomicTrie.TrieDB(),
 		vm.warpBackend,
 		vm.networkCodec,
+		vm.leafRequestTypeConfigs,
```

Comment:
could we return the map from `RegisterLeafRequestHandler` and perhaps rename it to `makeLeafRequestHandlers` to avoid a one-time use field on vm? seems like it doesn't need to be defined on vm

Reply:
This was in fact to use it from outside this pkg and let an external wrapping VM to register it's own handler.